File: pinyougou/pinyougou-sellergoods/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.PageResult;

import java.util.List;
import java.util.Map;

/**
 * Date2018/11/23
 * 品牌业务层接口
 */
public interface BrandService extends BaseService<TbBrand>{

    //自定义条件分页查询
    PageResult search(Integer page, Integer rows, TbBrand brand);

    //自定义获得品牌下拉列表
    List<Map<String,Object>> selectOptionList();
}

File: pinyougou/pinyougou-shop-web/src/main/webapp/js/service/brandService.js
// 服务层
app.service("brandService",function ($http) {

    // 分页查询全部列表,this代表BrandService
    this.findPage = function (page, rows) {
        return $http.get("../brand/findPage.do?page=" + page + "&rows=" + rows);
    };

    //新增
    this.add = function (eneity) {
        return $http.post("../brand/add.do",eneity);
    };

    //修改
    this.update = function (eneity) {
        return $http.post("../brand/update.do",eneity);
    };

    //根据品牌id查询品牌数据
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    };

    //执行删除
    this.delete=function (selectedIds) {
        return $http.get("../brand/delete.do?ids="+selectedIds)
    };

    //分页条件查询(搜索)
    this.search=function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page=" + page + "&rows=" + rows, searchEntity);
    };

    //获取格式化的品牌列表
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }
});